airstable.rb

```diff
@@ -0,0 +1,256 @@
+require 'sinatra'
+require './setup'
+require './models'
+require 'PP'
+
+set(:sessions, true)
+set(:session_secret, ENV['SESSION_SECRET'])
+
+# set(:requires_login) do
+#   condition do
+#     redirect '/' unless user_signed_in?
+#   end
+#   true
+# end
+
+# Before filter that redirects user to login page unless user is logged in
+# or page does not require login
+
+### This eliminates the possibility of a 404 page. Better way to redirect only
```

Comment:
Yeah, I linked to some examples in Slack before.  First, think about where you want the knowledge of whether to redirect or not to live.  It'd be better if it lives near the route definition, otherwise you create these two chunks of code that are separated physically and logically which have to be kept in sync / need implicit knowledge of each other.  Add a new route?  Don't forget to update this filter!  Change the logic in the filter?  Don't forget to make sure none of the routes depend on the old behavior!

https://codeunion.slack.com/archives/web-fundamentals/p1425078919000186